libs/utils/src/tallies.ts

```diff
@@ -0,0 +1,35 @@
+import { strict as assert } from 'assert'
+import {
+  ContestOptionTally,
+  ContestTally,
+  Dictionary,
+} from '@votingworks/types'
+
+export function combineContestTallies(
+  firstTally: ContestTally,
+  secondTally: ContestTally
+): ContestTally {
+  assert(firstTally.contest.id === secondTally.contest.id)
+  const combinedTallies: Dictionary<ContestOptionTally> = {}
+
+  for (const optionId of Object.keys(firstTally.tallies)) {
+    const firstTallyOption = firstTally.tallies[optionId]
+    assert(firstTallyOption)
+    const secondTallyOption = secondTally.tallies[optionId]
+    combinedTallies[optionId] = {
+      option: firstTallyOption.option,
+      tally: firstTallyOption.tally + (secondTallyOption?.tally || 0),
```

Comment:
I generally prefer `??` in such cases, though it doesn't _always_ make sense to use it over `||`:
```suggestion
      tally: firstTallyOption.tally + (secondTallyOption?.tally ?? 0),
```